src/dirwatcher/dirwatcher.js

```diff
@@ -0,0 +1,52 @@
+import fs from 'fs';
+import EventEmitter from 'events';
+class MyEmitter extends EventEmitter{};
+
+export class DirWatcher{
+    watch(path, delay){
+        this.myEmitter = new MyEmitter();
+        let files, files2;
+        setInterval(()=>{
+            files2 = fs.readdirSync(path);
+            if(files){
```

Comment:
I don't think this check is really good. It's a bad practice to initialize variables without default value assignment. I'd suggest you to make 
```
const files = [];
const files = [];
...
if (Array.isArray(files) && files.length) {}
```
So that in `this.compare` you won't have an error for checking `arr1.length` as you gonna check it everytime on array

Reply:
Updated. Thanks